app/WebParsing/UtsgJsonParser.hs

```diff
@@ -1,83 +1,95 @@
+-- | insert/update all the data into the Meeting and Times schema by creating and sending
+--   the http request to Artsci Timetable and then parsing the JSON response
 insertAllMeetings :: T.Text -> SqlPersistM ()
 insertAllMeetings org = do
     liftIO . print $ T.append "parsing JSON data from: " org
-    resp <- liftIO . simpleHttp $ T.unpack (T.append timetableApiUrl org)
-    let coursesLst :: Maybe (HM.HashMap T.Text (Maybe DB)) = decode resp
-        courseData = maybe [] (map dbData . catMaybes . HM.elems) coursesLst
-        -- courseData contains courses and sections;
-        -- only sections are currently stored here.
-        (_, sections) = unzip courseData
-        meetings = concat sections
-    mapM_ insertMeeting meetings
+
+    -- Define the request
+    let reqBody = object [ "campuses" .= ([] :: [T.Text])
+                                , "courseCodeAndTitleProps" .= object
+                                    [ "courseCode" .= ("" :: T.Text)
+                                    , "courseSectionCode" .= ("" :: T.Text)
+                                    , "courseTitle" .= org
+                                    , "searchCourseDescription" .= True
+                                    ]
+                                , "courseLevels" .= ([] :: [T.Text])
+                                , "creditWeights" .= ([] :: [T.Text])
+                                , "dayPreferences" .= ([] :: [T.Text])
+                                , "deliveryModes" .= ([] :: [T.Text])
+                                , "departmentProps" .= ([] :: [T.Text])
+                                , "direction" .= ("asc" :: T.Text)
+                                , "divisions" .= [T.pack "ARTSC"]
+                                , "instructor" .= ("" :: T.Text)
+                                , "page" .= (1 :: Int)
+                                , "pageSize" .= (200 :: Int)
+                                , "requirementProps" .= ([] :: [T.Text])
+                                , "sessions" .= [T.pack "20239", T.pack "20241", T.pack "20239-20241"]
+                                , "timePreferences" .= ([] :: [T.Text])
+                                ]
+        reqHeaders = [("Content-Type", "application/json"), ("Accept", "application/json")]
+    request <- liftIO $ parseRequest (T.unpack timetableApiUrl)
+    let request' = request { method = "POST"
+                           , requestBody = RequestBodyLBS $ encode reqBody
+                           , requestHeaders = reqHeaders
+                           }
+
+    -- Make the request
+    manager <- liftIO $ newManager tlsManagerSettings
+    response <- liftIO $ httpLbs request' manager
+    let respBody = responseBody response
+
+    -- decode the response
+    let meetings :: Maybe DBList = decode respBody
+    case meetings of
+        Nothing -> return ()
+        Just (DBList dbs) -> forM_ dbs $ \(DB meetTimes) -> mapM_ insertMeeting meetTimes
```

Comment:
This is a bit inelegant. You should create a helper function `DBList -> [MeetTime]` so that you can do a `mapM_ insertMeeting $ ... dbList`